packages/docregistry/src/context.ts

```diff
@@ -362,12 +362,31 @@ class Context<T extends DocumentRegistry.IModel> implements DocumentRegistry.ICo
     }
     let oldPath = change.oldValue && change.oldValue.path;
     let newPath = change.newValue && change.newValue.path;
-    if (newPath && oldPath === this._path) {
+
+    if (newPath && this._path.indexOf(oldPath) === 0) {
+      let changeModel = change.newValue;
+      // When folder name changed, `oldPath` is `foo`, `newPath` is `bar` and `this._path` is `foo/test`,
+      // we should update `foo/test` to `bar/test` as well
+      if (oldPath !== this._path) {
+        newPath = this._path.replace(new RegExp(`^${oldPath}`), newPath);
+        oldPath = this._path;
+        // Update client file model from folder change
+        changeModel = {
+          last_modified: change.newValue.created,
```

Comment:
Why use `newValue.created` instead of `newValue.last_modified`?

Reply:
I do want use last_modified time in here, but the data in newValue seems have other issue, here is the value

```json
{
    "mimetype": null,
    "writable": true,
    "name": "bbbb",
    "format": null,
    "created": "2018-06-21T03:51:19.938435Z",
    "content": null,
    "last_modified": "2018-06-21T03:51:13.902601Z",
    "path": "bbbb",
    "type": "directory",
    "size": null
}
```

last_modified was the create time of the folder, and created time was the rename time.

Maybe it was related with the new folder was actually recreated not renamed?

Reply:
Interesting, I'm not sure why that should be the case, but I also see that issue with the direct I also can't reproduce the error if the `last_modified` time is incorrect: I suspect that might be an OS-dependent thing.

Reply:
I tested in MacOS, I can reproduce the conflict issue later.